lib/internal/repl.js

```diff
@@ -92,7 +92,15 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
 
   function oninit(err, hnd) {
     if (err) {
-      return ready(err);
+      // Cannot open history file.
+      // Don't crash, just don't persist history.
+      debug('oninit: %s', err.message);
+      repl._writeToOutput('\nError: Could not open history file.\n' +
+                          'REPL session history will not be persisted.\n');
+      repl._refreshLine();
+      repl._historyPrev = _replHistoryMessage;
```

Comment:
I think this is already handled by https://github.com/nodejs/node/pull/3630/files#diff-ffb9ad69f91d159c68e827d213fb3064R68 ?